user/tests/wiring/power_management_long_running/power.cpp

```diff
@@ -0,0 +1,379 @@
+/*
+ * Copyright (c) 2021 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "application.h"
+#include "unit-test/unit-test.h"
+
+#if !HAL_PLATFORM_POWER_MANAGEMENT
+#error "Unsupported platform"
+#endif // !HAL_PLATFORM_POWER_MANAGEMENT
+
+SYSTEM_MODE(SEMI_AUTOMATIC);
+UNIT_TEST_APP();
+
+#define SERIAL      Serial1
+
+namespace {
+
+constexpr uint32_t DISCHARGE_STATE1 = 0xc00f;
+constexpr uint32_t DISCHARGE_STATE2 = 0xd00f;
+constexpr uint32_t CHARGING_STATE_DISABLED_CHARGING = 0xbeef;
+constexpr uint32_t CHARGING_STATE_ENABLED_CHARGING = 0xdeed;
+constexpr uint32_t CHARGED_STATE = 0xc0fe;
+constexpr uint32_t DISCONNECTED_STATE = 0xd00d;
+constexpr uint32_t DISCONNECTED_STATE_SLEEP = 0xbead;
+constexpr uint32_t CONNECTED_STATE_SLEEP = 0xb007;
+uint32_t g_state;
```

Comment:
Nit: we normally do this when we need to keep state between the resets as a retained variable. Here it's used only in a single place as a conditional, so perhaps we can just take special care with that specific test case.

Reply:
It is a global variable being used in every test case to track the testing state, if certain test case fails, then the following test cases will be skipped.